lib/postgrex/protocol.ex

```diff
@@ -23,7 +23,8 @@ defmodule Postgrex.Protocol do
             queries: nil,
             postgres: :idle,
             transactions: :strict,
-            buffer: nil
+            buffer: nil,
+            disconnect_on_read_only_transaction_error: false
```

Comment:
I'm not a fan either of this option name or putting it on the state but can't think of anything better at the moment.

Reply:
I think both of these are fine! :+1:

Reply:
@fishcakez I like making it a bit more general. :+1: it should also address @wojtekmach's concern with naming. :)

Reply:
sounds great, how about `:disconnect_on_errors`?

Reply:
@wojtekmach excellent.